src/mappings/utils/db.ts

```diff
@@ -140,12 +142,23 @@ export async function optimizedBulkCreate(modelName: string, docs: Array<any>, t
       `[optimizedBulkCreate] Processing batch: Start index = ${batchStart}, Records in batch = ${records.length}, Remaining records = ${docs.length - batchStart - records.length}.`,
     );
 
-    // Perform the bulk creation
-    await store.modelProvider.getModel(modelName).model.bulkCreate(records, {
-      transaction: store.context.transaction,
-      ignoreDuplicates: true,
-      ...bulkCreateOpts,
-    });
+    // TODO(@Alann27): Instead of creating a transaction for each batch, we should create a transaction for multiple bashes
+    const transaction = await model.sequelize!.transaction({
+      logging: false
+    })
+
+    try {
+      // Perform the bulk creation
+      await model.bulkCreate(records, {
+        transaction,
+        ignoreDuplicates: true,
+        ...bulkCreateOpts,
+      });
+
+      await transaction.commit()
+    } catch {
+      await transaction.rollback()
```

Comment:
Here you miss to bubble the error to break this, otherwise this will keep going leading to missing data.